sql/core/src/test/resources/sql-tests/results/typeCoercion/native/promoteStrings.sql.out

```diff
@@ -198,15 +198,16 @@ SELECT '1' - cast('2017-12-11 09:30:00.0' as timestamp) FROM t
 struct<>
 -- !query output
 org.apache.spark.sql.AnalysisException
-cannot resolve '('1' - CAST('2017-12-11 09:30:00.0' AS TIMESTAMP))' due to data type mismatch: argument 1 requires timestamp type, however, ''1'' is of string type.; line 1 pos 7
+cannot resolve '('1' - CAST('2017-12-11 09:30:00.0' AS TIMESTAMP))' due to data type mismatch: differing types in '('1' - CAST('2017-12-11 09:30:00.0' AS TIMESTAMP))' (string and timestamp).; line 1 pos 7
 
 
 -- !query
 SELECT '1' - cast('2017-12-11 09:30:00' as date)        FROM t
 -- !query schema
-struct<(1 - CAST(2017-12-11 09:30:00 AS DATE)):day-time interval>
+struct<>
 -- !query output
-NULL
+org.apache.spark.sql.AnalysisException
```

Comment:
@cloud-fan @gengliangwang Are we ok to change the behavior?

Reply:
This doesn't look good...Why does that happen?

Reply:
Probably because we have special type coercion logic for BinaryOperators?